### PR TITLE
remarshal 0.3.0

### PR DESCRIPTION
**Pull request by dai0304**

### CODE REVIEW

File: Library/Formula/remarshal.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require "formula"
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]: :scissors: this line

File: Library/Formula/remarshal.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class Remarshal < Formula
+  homepage "https://gowalker.org/github.com/dbohdan/remarshal"
+  url 'https://github.com/dbohdan/remarshal.git', :tag => "v0.3.0", :revision => 'bb9ac467ca297c55587a798531886e6159acbcbd'
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]: double quotes everywhere.

File: Library/Formula/remarshal.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    system "go", "build", "-o", "remarshal"
+    bin.install 'remarshal'
+
+     Array["toml", "yaml", "json"].each do |informat|
+      Array["toml", "yaml", "json"].each do |outformat|
+        bin.install_symlink "remarshal" => "#{informat}2#{outformat}"
+      end
+    end
+  end
+end
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]: Can you write a test case? Thanks.

File: Library/Formula/remarshal.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    # Install Go dependencies
+    system "go", "get", "github.com/BurntSushi/toml"
+    system "go", "get", "gopkg.in/yaml.v2"
+
+    # Build and install
+    system "go", "build", "-o", "remarshal"
+    bin.install 'remarshal'
+
+     Array["toml", "yaml", "json"].each do |informat|
+      Array["toml", "yaml", "json"].each do |outformat|
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]:
```ruby
["toml", "yaml", "json"].combination(2).each do | informat, outformat|
```

File: Library/Formula/remarshal.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  depends_on "go" => :build
+  depends_on :hg => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    # Install Go dependencies
+    system "go", "get", "github.com/BurntSushi/toml"
+    system "go", "get", "gopkg.in/yaml.v2"
+
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]: You need to use `go_resource`. See `aptly.rb` as example.

**Comment by dai0304** [comment]
All findings by @xu-cheng are fixed.  Thanks.

File: Library/Formula/remarshal.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
+require 'language/go'
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]: Sorry but I mean double quotes everywhere not simple quote.

File: Library/Formula/remarshal.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    end
+  end
+
+  test do
+    require 'open3'
+    Open3.popen3("#{bin}/remarshal", '-if=json', '-of=yaml') do |stdin, stdout, _|
+      stdin.write('{ "foo.bar": "baz", "qux": 1 }')
+      stdin.close
+      assert_equal "foo.bar: baz\nqux: 1\n\n", stdout.read
+    end
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]: You can use `pipe_output` Instead of open3

File: Library/Formula/remarshal.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
+require "language/go"
+
+class Remarshal < Formula
+  homepage "https://gowalker.org/github.com/dbohdan/remarshal"
```
Comment (xu-cheng) [inline Library/Formula/remarshal.rb]: The homepage need to be changed to https://github.com/dbohdan/remarshal". Thanks.

**Comment by xu-cheng** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support.

**Pull request closed without merging**